gax/src/main/java/com/google/api/gax/core/GaxProperties.java

```diff
@@ -57,9 +57,13 @@ public static String getLibraryVersion(Class<?> libraryClass) {
    * method are expected to be cached.
    */
   public static String getLibraryVersion(Class<?> libraryClass, String propertyName) {
-    String version = getLibraryVersion(libraryClass);
-    if (!DEFAULT_VERSION.equals(version)) {
-      return version;
+    String version = null;
+    // Always read GaxProperties' version from the properties file.
+    if (!libraryClass.equals(GaxProperties.class)) {
+      version = getLibraryVersion(libraryClass);
+      if (!DEFAULT_VERSION.equals(version)) {
+        return version;
+      }
     }
 
     try (InputStream in = libraryClass.getResourceAsStream("/dependencies.properties")) {
```

Comment:
This reads `dependencies.properteis`, but `generateProjectProperties` writes to the file with typo `dependencies.propertoes`. Looks like the file (even if it did not contain a typo) is never read (and the tests pass without the need to copy the file).

Reply:
Fixed.